ethexe/processor/src/tests.rs

```diff
@@ -1232,14 +1249,279 @@ async fn injected_prioritized_over_canonical() {
     processor.process_queue(&mut handler).await;
 
     let to_users = handler.transitions.current_messages();
-    assert!(to_users.len() <= MSG_NUM);
-
-    // Verify that canonical messages were not processed
-    for (idx, (_, message)) in to_users.into_iter().enumerate() {
-        assert_eq!(message.destination, injected_user);
-        // Skip first message which is INIT reply
-        if idx != 0 {
-            assert_eq!(message.payload, b"PONG");
+
+    // Verify that injected messages were processed first
+    // (skip first message which is INIT reply)
+    let mut is_canonical_found = false;
+    for (_, message) in to_users.into_iter().skip(1) {
+        if message.destination == canonical_user {
+            is_canonical_found = true;
+        } else if is_canonical_found && message.destination == injected_user {
+            panic!("Canonical message processed before injected one");
         }
     }
 }
+
+#[tokio::test(flavor = "multi_thread")]
+async fn executable_balance_charged() {
+    init_logger();
+
+    let mut processor = Processor::new(Database::memory()).unwrap();
```

Comment:
why wouldn't you wrap this boilerplate in some fn ? so tests will be focused on exact scope